### PR TITLE
Made cards display images, made cards display in random order, made cards display their location

### PR DESCRIPTION
**Pull request by rako77**

### CODE REVIEW

File: src/cards.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ export interface Expansion {
 export interface Card {
     name: string;
     set: Expansion;
+    image: string;
```
Comment (shocklateboy92) [inline src/cards.ts:11]: `imageUrl` please

File: src/cards.ts [hunk 2]
```diff
@@ -19,11 +21,33 @@ const sets = CardInfo.reduce(
     {} as Dictionary<Expansion>
 );
 
-const cards: Card[] = CardInfo.map(card => ({
+const validCards = CardInfo.filter(card => (
+    card.set !== "Dark Ages" && 
+    card.type === "kingdom" && 
+    !(card.set === "Promo" && card.name !== "Walled Village" && card.name !== "Stash") &&
```
Comment (shocklateboy92) [inline src/cards.ts:27]: please do something like `!PROMO_CARDS.includes(card.name)`

File: src/cards.ts [hunk 3]
```diff
@@ -19,11 +21,33 @@ const sets = CardInfo.reduce(
     {} as Dictionary<Expansion>
 );
 
-const cards: Card[] = CardInfo.map(card => ({
+const validCards = CardInfo.filter(card => (
+    card.set !== "Dark Ages" &&
```
Comment (shocklateboy92) [inline src/cards.ts:25]: extract a constant like `MISSING_SETS`

File: src/cards.ts [hunk 4]
```diff
@@ -19,11 +21,33 @@ const sets = CardInfo.reduce(
     {} as Dictionary<Expansion>
 );
 
-const cards: Card[] = CardInfo.map(card => ({
+const validCards = CardInfo.filter(card => (
+    card.set !== "Dark Ages" && 
+    card.type === "kingdom" &&
```
Comment (shocklateboy92) [inline src/cards.ts:26]: constant please

File: src/cards.ts [hunk 5]
```diff
@@ -19,11 +21,33 @@ const sets = CardInfo.reduce(
     {} as Dictionary<Expansion>
 );
 
-const cards: Card[] = CardInfo.map(card => ({
+const validCards = CardInfo.filter(card => (
```
Comment (shocklateboy92) [inline src/cards.ts:24]: Try doing the `filter()` on the `cards` array (that has already gone through `map()`) once.

**Review by shocklateboy92 — CHANGES_REQUESTED** [review]

**Comment by shocklateboy92** [comment]
Also, please format the files:
`ext install esbenp.prettier-vscode`

File: src/cards.ts (hunk 3 above)
Comment (ghost) [reply src/cards.ts:25]: irrelevant, this will change as we would probably need to do a way to select the sets you have

File: src/cards.ts (hunk 2 above)
Comment (ghost) [reply src/cards.ts:27]: yes but this will change.

File: src/cards.ts [hunk 6]
```diff
@@ -19,11 +21,33 @@ const sets = CardInfo.reduce(
...
+    card.set !== "Dark Ages" && 
+    card.type === "kingdom" && 
+    !(card.set === "Promo" && card.name !== "Walled Village" && card.name !== "Stash") && 
+    !(card.set==="Intrigue 2nd Edition" && card.status === "removed")
```
Comment (ghost) [inline src/cards.ts:28]: like @shocklateboy92  said all of these strings need to be constants, but I don't think it should be blocking as of now.

File: src/components/card-display.tsx [hunk 7]
```diff
@@ -3,11 +3,11 @@ import * as React from 'react';
 
 export const CardDisplay: React.FunctionComponent<{
     card: Card;
-    imageUrl: string | undefined;
 }> = props => (
     <div className='card float'>
         {props.card.name}
-        <img src={props.imageUrl} />
+        <img src={props.card.image} />
+        <br/>{`Box: ${Math.floor(props.card.index/210)+1}, Row: ${Math.floor(props.card.index%210/30)+1}, Column: ${props.card.index%30+1}`}
```
Comment (ghost) [inline src/components/card-display.tsx:10]: Reverse the order to Column: X Row: Y

**Pull request closed without merging**